lib/services/local/headlessauthn_watcher.go

```diff
@@ -203,14 +207,14 @@ func (h *HeadlessAuthenticationWatcher) watch(ctx context.Context) error {
 	}
 }
 
-func (h *HeadlessAuthenticationWatcher) notify(headlessAuthns ...*types.HeadlessAuthentication) {
+func (h *HeadlessAuthenticationWatcher) notify(putEvent ...backend.Item) {
 	h.mux.Lock()
 	defer h.mux.Unlock()
-	for _, headlessAuthn := range headlessAuthns {
+	for _, item := range putEvent {
 		for i := range h.waiters {
-			if h.waiters[i].name == headlessAuthn.Metadata.Name {
+			if h.waiters[i].key == string(item.Key) {
 				select {
-				case h.waiters[i].ch <- headlessAuthn:
+				case h.waiters[i].ch <- item:
```

Comment:
The marshal/unmarshal dance seems a bit strange, I kind of prefer the stronger types we had before.

Would it help/simplify if you did `proto.Clone(headlessAuthn).(*types.HeadlessAuthentication)` before sending the item to the waiter?

Reply:
Yeah that's a fair point, I'll modify it